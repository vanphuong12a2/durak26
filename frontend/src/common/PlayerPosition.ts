export enum PlayerPosition {
  BOTTOM = 'BOTTOM',
  TOP = 'TOP',
  RIGHT = 'RIGHT',
  LEFT = 'LEFT',
}

export const allPlayerPositions = [PlayerPosition.BOTTOM, PlayerPosition.RIGHT, PlayerPosition.TOP, PlayerPosition.LEFT];
