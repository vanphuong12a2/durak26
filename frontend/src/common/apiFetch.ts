const apiFetch = (url: string, params: any = {}) =>
  fetch(url, params)
    .then(response => {
      if (response.ok) {
        return response;
      } else {
        throw new Error('Status code is not 200');
      }
    })
    .then(res => res.json());

export default apiFetch;
