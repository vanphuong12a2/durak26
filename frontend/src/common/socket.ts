import socketIOClient from 'socket.io-client';
import sailsIOClient from 'sails.io.js';

const io = sailsIOClient(socketIOClient);

if (process.env.NODE_ENV !== 'production') {
  io.sails.url = 'http://localhost:1337';
}

const socket = io.socket;

export default socket;
