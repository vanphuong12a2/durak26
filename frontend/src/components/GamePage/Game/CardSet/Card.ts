export interface Card {
  id: string
  isUp: boolean
}
