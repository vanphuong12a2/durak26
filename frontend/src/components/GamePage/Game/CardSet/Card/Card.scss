.card {
  margin: 0;
  max-width: 80px;
}
