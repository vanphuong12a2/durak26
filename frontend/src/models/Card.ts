class Card {
  public value: string;
  public suit: string;

  constructor(value: string, suit: string) {
    this.value = value;
    this.suit = suit;
  }
}

export default Card;
