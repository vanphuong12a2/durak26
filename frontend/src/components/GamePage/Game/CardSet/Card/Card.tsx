import React from 'react';
import * as cards from '../../../../../images/index';
import CardData, {UNKNOWN_VALUE} from '../../../../../models/CardData';
import './Card.scss';

interface Props {
  card: CardData
}

const Card = (props: Props) => {
  if(props.card.value === UNKNOWN_VALUE) {
    return (
      <img className='card' src={cards.back}/>
    )
  } else {
    // @ts-ignore
    const cardSvg = cards[`_${props.card.value}${props.card.suit}`];
    return (
      <img className='card' src={cardSvg}/>
    );
  }
};

export default Card;
