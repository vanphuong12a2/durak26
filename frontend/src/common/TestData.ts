import createMockStore, {MockStoreCreator, MockStoreEnhanced} from 'redux-mock-store';
import {ApplicationState} from '../store/ApplicationState';
import thunk, {ThunkDispatch} from 'redux-thunk';
import {AnyAction} from 'redux';
import PlayerData, {PlayerPosition} from '../models/PlayerData';
import CardData, {UNKNOWN_VALUE} from '../models/CardData';
import {CardPosition, CardsMap} from '../models/CardsMap';

export const aCard = new CardData('2', 'D');
export const anUnknownCard = new CardData(UNKNOWN_VALUE, UNKNOWN_VALUE);
export const aPlayer = new PlayerData('someId', 'some-name', PlayerPosition.BOTTOM);

type DispatchExts = ThunkDispatch<ApplicationState, undefined, AnyAction>
const mockStoreCreator: MockStoreCreator<ApplicationState, DispatchExts> =
  createMockStore<ApplicationState, DispatchExts>([thunk]);

export const cards: CardsMap = {[CardPosition.TO_DEAL]: [anUnknownCard, anUnknownCard]};

const getInitialApplicationState = () => (
  {
    game: {
      loading: false,
      newGame: {
        loading: false
      }
    },
    card: {
      cards
    },
    player: {
      players: [],
      newPlayer: {
        loading: false
      }
    }
  }
);

export const freshTestStore = () => mockStoreCreator(getInitialApplicationState());
export const testStore: MockStoreEnhanced<ApplicationState, DispatchExts> = freshTestStore();
