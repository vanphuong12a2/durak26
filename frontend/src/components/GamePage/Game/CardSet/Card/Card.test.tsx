import React from 'react';
import {create} from 'react-test-renderer';
import {FaceDownCard, FaceUpCard} from './Card';
import {aCard} from '../../../../../common/TestData';

describe('<FaceUpCard />', () => {

  describe('Snapshots', () => {
    it('renders face-up card correctly', () => {
      const component = create(<FaceUpCard card={aCard}/>);
      expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders face-down card correctly', () => {
      const component = create(<FaceDownCard/>);
      expect(component.toJSON()).toMatchSnapshot();
    });
  });

});
