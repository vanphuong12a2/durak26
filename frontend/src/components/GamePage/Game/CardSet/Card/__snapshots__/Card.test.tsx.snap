// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Snapshots renders face-down card correctly 1`] = `
<img
  className="card"
  src="card_back.svg"
/>
`;

exports[`<Card /> Snapshots renders face-up card correctly 1`] = `
<img
  className="card"
  src="2D.svg"
/>
`;
